pkg/yttlibrary/overlay/insert_annotation.go

```diff
@@ -36,20 +49,23 @@ func NewInsertAnnotation(newItem template.EvaluationNode) (InsertAnnotation, err
 		kwargName := string(kwarg[0].(starlark.String))
 
 		switch kwargName {
-		case "before":
+		case InsertAnnotationKwargBefore:
 			resultBool, err := tplcore.NewStarlarkValue(kwarg[1]).AsBool()
 			if err != nil {
 				return InsertAnnotation{}, err
 			}
 			annotation.before = resultBool
 
-		case "after":
+		case InsertAnnotationKwargAfter:
 			resultBool, err := tplcore.NewStarlarkValue(kwarg[1]).AsBool()
 			if err != nil {
 				return InsertAnnotation{}, err
 			}
 			annotation.after = resultBool
 
+		case InsertAnnotationKwargVia:
+			annotation.via = &kwarg[1]
+
```

Comment:
_(Commentary, not a request for a change, per se.)_

It's interesting that in `@overlay/replace` (and therefore here) we store the `via=` as a `starlark.Value` (rather than a `starlark.Callable`).
We also don't vet the argument, here — a natural choke-point for invalid input.

The advantage of vetting/constraining as early as possible is that it _generally_ simplifies subsequent logic, reducing complexity overall.

Since there's very little distance between `NewAssertionAnnotation()` and `Value()` (below), I don't see that complexity reduction: just moving code around.

It's more meaningful as aiming for exemplary code (i.e. worthy of imitation).

It's not the responsibility of the author of this PR to "fix" this; nor is this a blocker for this PR (that's impractical).

Reply:
I have refactored both annotations to store their `via` kwarg as a `starklark.Callable`. To avoid regression, the error case for both are covered with a template test.

My first intuition was to extend `pkg.template.core.StarlarkValue` with a `AsCallable`, but that does not work because Go has no function type that is arg-agnostic, which is why we like Go. And so, I settled for a straight-forward type assertion within the `New*` constructor of each annotation.